runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -2401,6 +2401,12 @@ static void jitHookClassLoaderUnload(J9HookInterface * * hookInterface, UDATA ev
 
    TR::CompilationInfo * compInfo = TR::CompilationInfo::get(jitConfig);
 
+   if (compInfo->getTransientClassLoadersSet())
+     {
+     compInfo->getTransientClassLoadersSet()->erase(classLoader);
```

Comment:
Let's add a comment saying that locks are not required because the compilation threads are stopped when class unloading happens.